src/data-table/index.js

```diff
@@ -149,18 +148,51 @@ class DataTable extends Component {
     }
   }
 
+  replaceDiacritics = (str) => {
```

Comment:
This function can be a pure static function placed outside the `DataTable` class since it doesn't interact with class state or props. A quick and lazy way to do it is to place it right before the class definition.

Reply:
* Furthermore, you can even pull `diacritics` out of the function itself as an independent variable.
* To improve the runtime of the function `replaceDiacritics()`, `diacritics` can be set as a object in this case, like:
```
const diacritics = {
  'A': /[\300-\306]/g,
  'a': /[\340-\346]/g,
  ...
}
```
and how to use it:
```
str = str.split('').map(letter => diacritics[letter] || letter).join()
```
In this way, we avoid going through the string for each letter in `diacritics`(14 letters) and only go through it once.
Although we reduce its runtime to 1/14 by doing so, the difference is still unnoticeable.
However I think it's still a good practice to do and we want to make the search as fast as possible.